### ISSUE TITLE
what's the source of alpha equation

### ISSUE THREAD

**Issue by wx666-hub** [fmuni/boilingFoam-PUBLIC#7]
https://github.com/fmuni/boilingFoam-PUBLIC/blob/1a19028c1b9289971e09b1e00bf94d81bb6d936a/applications/solvers/icoBoilingFoam/fluid/VoF/alphaEqn.H#L120 .su and sp are written here as alpha formula source terms, but in the previous file alphaSuSp.H, it seems that Su and Sp are only initialized to 0, but when the phase transition exists, he should not be 0, may I ask where I understand wrong

**Comment by linxiayueguang** [comment]
I am equally puzzled by this.

**Comment by fmuni** [comment]
The semi-implicit source therm is (Sp + divU)*alpha. divU accounts for the change in volume caused by boiling at the interface.

**Comment by wx666-hub** [comment]
Thank for your answer, but the explicit source therm is (me+mc)/ρ1，if semi-implicit its α*(1/ρ1 - α*(1/ρ1-1/ρ2))*(me/α-mc/(1-α))+ α*(divU)+(1/ρ1 - α*(1/ρ1-1/ρ2))*mc/(1-α), because divU =( me+mc)/(1/ρ1-1/ρ2)

**Comment by fmuni** [comment]
You are not wrong, but there is usually more than one algorithm for solving a given problem. If you look at other parts of the code, you are going to find similar oddities that apparently don’t make sense. Taken alone and confronted with equations and procedures written in published papers, they just look like mistakes. But when you look at the whole algorithm, everything fits and you understand the reason behind certain choices. Many expressions end up looking incredibly similar or convenient when you run an order of magnitude analysis.

**Comment by wx666-hub** [comment]
OK, I'll further explain a similar theory. The solver you proposed has a significant increase in time step size when solving micro-channel flow boiling with the same Co. This might be due to the processing method of the source term. Could you provide any theoretical articles for this set of equations? Thank you very much!

**Comment by fmuni** [comment]
Yes, essentially you get higher time step for the same Co. I don't have an article that explains the algorithm, but the equations can be found in the paper of Hardt & Wondra (10.1016/j.jcp.2008.02.020). Equation 8 is essentially what generates the pressure equation and is the reason why I only have divU and not Sp or Su in the alpha equation. Another article that describes the equations is that of Andrea Ferrari (https://doi.org/10.1016/j.ijheatmasstransfer.2018.03.012), who wrote the code that motivated the development of boilingFoam.
